### PR TITLE
Fix/`PartialEq::ne`

### LINKED ISSUE #1: cannot call non-const fn `<Self as PartialEq<Rhs>>::eq` in constant functions
I'm trying to compile `const_units` using nightly but I keep getting this error:

> error[E0015]: cannot call non-const fn `<Self as PartialEq<Rhs>>::eq` in constant functions
>    --> /home/elpiel/.cargo/registry/src/index.crates.io-6f17d22bba15001f/const_ops-0.2.2/src/lib.rs:117:15
>     |
> 117 |         !self.eq(other)
>     |               ^^^^^^^^^
>     |
>     = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
>     = help: add `#![feature(effects)]` to the crate attributes to enable
> 
> For more information about this error, try `rustc --explain E0015`.
> error: could not compile `const_ops` (lib) due to 1 previous error
> 

I tried adding `#![feature(effects)]` to const_units but didn't work. Maybe it should be added to this crate instead?

### PR DESCRIPTION
**Pull request by elpiel**
Fixes #1

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,7 +1,6 @@
-#![feature(const_trait_impl, const_mut_refs, const_fn_floating_point_arithmetic)]
+#![feature(const_trait_impl, const_mut_refs, const_fn_floating_point_arithmetic, effects)]
 
 mod impls;
-pub use impls::*;
```
Comment (elpiel) [inline src/lib.rs:4]: not actually needed and was complaining that imports are not used.

Comment (haennes) [reply src/lib.rs:4]: i would like to leave this in, as this is needed to be compatible with the core / std implementation. (see README)
(Or am I wrong?)

File: rust-toolchain.toml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[toolchain]
+channel = "nightly"
+# components = [ "rustfmt", "rustc-dev" ]
```
Comment (haennes) [inline rust-toolchain.toml]: please remove those lines as they are commented out

**Comment by haennes** [comment]
Hey,

first of all.
Thanks for engaging in this project.
This has been laying around for a couple of months somewhat awaiting the approval of iliekturles.
I am however motivated to keep working on this. (to get started you may first take a look at the code generator)
Documentation is really bad sadly. Maybe i will have time to write some tomorrow

File: src/lib.rs (hunk 1 above)
Comment (elpiel) [reply src/lib.rs:4]: Not sure what you mean by compatible.
Can you elaborate?

image

File: rust-toolchain.toml (hunk 2 above)
Comment (elpiel) [reply rust-toolchain.toml]: removed

**Comment by elpiel** [comment]
No worries I'm always happy to help with something I am quite interested in.

I tried building the units but the build script never ends. Left it for a few minutes but nothing happened.

I will be giving a talk at Rust Nation and I would like to include the const generics version of UOM.
Hopefully, we can get more contributors but also make some awareness in regards to the nightly features that we still need stabilized.

**Comment by elpiel** [comment]
Ps: Need one more fix for one impl, I'll open a PR.

**Comment by haennes** [comment]
> No worries I'm always happy to help with something I am quite interested in.
> 
> I tried building the units but the build script never ends. Left it for a few minutes but nothing happened.
> 
> I will be giving a talk at Rust Nation and I would like to include the const generics version of UOM. Hopefully, we can get more contributors but also make some awareness in regards to the nightly features that we still need stabilized.

Great :) 
It would be great if we get a compiling version until then.....
Could you specify which talk in paticular. Couldnt find it :facepalm: 
I am a huge fan of these nightly features in paticular, as they are a game changer for many applications... Hopefully they will get stable soon :)
On that note: Do you know of any efforts to support generic const generics? (see  [here](https://github.com/iliekturtles/uom/issues/381#issuecomment-1790817370))

**Comment by elpiel** [comment]
>It would be great if we get a compiling version until then.....
Yes, I really want to get this up and running but lately with my schedule I'd try to do the best I can to prepare the crate for the talk, otherwise I can always mention it.

> Could you specify which talk in paticular. Couldnt find it 🤦
The talk title is "Typed for safety"

> On that note: Do you know of any efforts to support generic const generics? (see https://github.com/iliekturtles/uom/issues/381#issuecomment-1790817370)

No idea to be honest...

**Comment by haennes** [comment]
great.

any opinions on the discussion above about the
 ```rust
 use impls
```

**Comment by elpiel** [comment]
> great.
> 
> any opinions on the discussion above about the
>  ```rust
>  use impls
> ```

It should not be needed. An implemented trait on a type is just that, there's nothing to export.
Let's leave it as it is for now. It can always be removed.

**Pull request merged**